### PR TITLE
[OYPD-228] Custom logo can't be set for footer.

### PR DESCRIPTION
**Pull request by Kerby70**
Jira https://propeople-us.atlassian.net/browse/OYPD-228
D.o https://www.drupal.org/node/2854173

Steps to review:
- [x] Login as admin
- [x] Go to /admin/appearance/settings/openy_rose
- [x] Add an image to the Footer Logo field
- [x] Submit and see message "Footer Logo saved."
- [x] Go to homepage
- [x] See new image as footer logo.

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>6</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build513/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build513

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/513/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build513/profiles/openy/build/reports/behat

File: themes/openy_themes/openy_rose/openy_rose.theme [hunk 1]
```diff
@@ -104,3 +104,47 @@ function openy_rose_preprocess_field__node__field_location_directions(&$variable
   // Change link title to Directions.
   $variables['items'][0]['content']['#title'] = t('Directions');
 }
+
+function openy_rose_form_system_theme_settings_alter(&$form, \Drupal\Core\Form\FormStateInterface &$form_state, $form_id = NULL) {
```
Comment (ddrozdik) [inline themes/openy_themes/openy_rose/openy_rose.theme]: Please add comments to this function.

File: themes/openy_themes/openy_rose/openy_rose.theme [hunk 2]
```diff
@@ -104,3 +104,47 @@ function openy_rose_preprocess_field__node__field_location_directions(&$variable
...
+        $file->setPermanent();
+        $file->save();
+        // Sets a Drupal message just for clarity .
+        drupal_set_message(t('Footer Logo saved.'), 'status');
+      }
+    }
+  }
+}
+
+function openy_rose_preprocess_page(&$variables) {
```
Comment (ddrozdik) [inline themes/openy_themes/openy_rose/openy_rose.theme]: Please add comments to this function.

**Review by ddrozdik — CHANGES_REQUESTED** [review]

**Comment by ddrozdik** [comment]
Also please fix sniffer issues

**Comment by ddrozdik** [comment]
- I suggest to move this field into the fieldset named "Logo image".
- svg is not allowed, I think we should allow upload svg. 
- would be good to mention suggested dimensions for the image in the description. 
- I think we can remove message "Footer Logo saved." since there is a message "The configuration options have been saved."

Questions:
- Are we going to have default logo image in the footer?

**Comment by alexschedrov** [comment]
@ddrozdik yes, we should be able to upload SVG.
Here's an example of settings form on YGTC https://github.com/propeoplemd/ymca/blob/master/docroot/themes/custom/ymca/ymca.theme#L1460

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>13</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build516/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build516

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/516/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build516/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>12</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build517/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build517

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/517/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build517/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build520

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/520/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build520/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build521

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/521/

**Comment by alexschedrov** [comment]
1. Please display default logo from theme when there's no logo specified.

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build521/profiles/openy/build/reports/behat

**Comment by Kerby70** [comment]
@ddrozdik I have addressed each of the issues. @Sanchiz answered the default logo question above.

Thank you both.

**Review by alexschedrov — CHANGES_REQUESTED** [review]
Default footer logo should be displayed

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build524

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/524/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build524/profiles/openy/build/reports/behat

File: themes/openy_themes/openy_rose/openy_rose.theme [hunk 3]
```diff
@@ -7,11 +7,16 @@
 
 use Drupal\Core\Render\Element;
 use Drupal\Core\Render\Element\RenderElement;
+use \Drupal\Core\Form\FormStateInterface;
+use \Drupal\file\Entity\File;
 
 /**
  * Implements hook_preprocess_node().
+ *
+ * @param array $variables
```
Comment (ddrozdik) [inline themes/openy_themes/openy_rose/openy_rose.theme]: No need to list params in comments for hooks.

File: themes/openy_themes/openy_rose/openy_rose.theme [hunk 4]
```diff
@@ -20,8 +25,11 @@ function openy_rose_preprocess_node(&$variables) {
 
 /**
  * Implement hook_preprocess_paragraph().
+ *
+ * @param array $variables
```
Comment (ddrozdik) [inline themes/openy_themes/openy_rose/openy_rose.theme]: No need to list params in comments for hooks.

Comment (ddrozdik) [reply themes/openy_themes/openy_rose/openy_rose.theme]: and same issue below ...

File: themes/openy_themes/openy_rose/templates/page/page.html.twig [hunk 5]
```diff
@@ -152,7 +152,10 @@
     <footer class="footer">
       <div class="container">
         <div class="col-sm-2 col-xs-4 hidden-sm hidden-md">
-          <a href="{{ base }}" class="footer__brand brand" title="{{ "YMCA"|t }}"></a>
+          {% if footer_logo %}
+            {% set background_image = "style='background-image: url(" ~ file_url(footer_logo) ~ ")'" %}
+          {% endif %}
+          <a href="{{ base }}" class="footer__brand brand" title="{{ "YMCA"|t }}" {{ background_image|raw }}></a>
```
Comment (ddrozdik) [inline themes/openy_themes/openy_rose/templates/page/page.html.twig]: If variable `footer_logo` is undefined then `background_image ` is undefined as well. You need to check this before using variable.

**Review by ddrozdik — CHANGES_REQUESTED** [review]

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build550

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/550/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build550/profiles/openy/build/reports/behat

**Review by ddrozdik — APPROVED** [review]

**Comment by ddrozdik** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build555

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/555/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build555/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]

**Pull request merged**